api-common/src/main/java/org/openmrs/module/metadatasharing/handler/impl/ConceptMergeHandler.java

```diff
@@ -56,6 +57,19 @@ public void merge(Concept existingConcept, Concept incomingConcept, ImportType i
 							}
 						}
 					}
+
+					// Update name capitalization to avoid creating duplicates, if necessary
+					Map<String, ConceptName> existingLowerCaseNames = new HashMap<String, ConceptName>();
+					for (ConceptName existingName : existingConcept.getNames(locale)) {
+						existingLowerCaseNames.put(existingName.getName().toLowerCase(), existingName);
+					}
+					for (ConceptName incomingName : incomingConcept.getNames(locale)) {
+						ConceptName existingName = existingLowerCaseNames.get(incomingName.getName().toLowerCase());
+						if (existingName != null && !incomingName.getName().equals(existingName.getName())) {
+							existingName.setName(incomingName.getName());
+							incomingConcept.removeName(incomingName);
```

Comment:
What happens if we do not remove `incomingName` from `incomingConcept`?

Reply:
I don't know. I would assume that since the incoming name would now exactly match an existing name that it wouldn't migrate it. However, there's the possibility that the incoming name has a different UUID that the one in the existing concept, in which case I think the duplicate name would be created, causing an error. Therefore I guess this line makes it a bit more robust in this particular case.

Are you concerned about it? Should I investigate?

Reply:
What about if something else differs, like the locale, name type, etc?